crates/parcel_core/src/types/dependency.rs

```diff
@@ -157,6 +158,40 @@ impl Dependency {
     self.hash(&mut hasher);
     hasher.finish()
   }
+
+  pub fn set_placeholder(&mut self, placeholder: impl Into<serde_json::Value>) {
+    self.meta.insert("placeholder".into(), placeholder.into());
+  }
+
+  pub fn set_is_webworker(&mut self) {
+    self.meta.insert("webworker".into(), true.into());
+  }
+
+  pub fn set_kind(&mut self, kind: impl Into<serde_json::Value>) {
+    self.meta.insert("kind".into(), kind.into());
+  }
+
+  pub fn set_should_wrap(&mut self, should_wrap: bool) {
+    self.meta.insert("shouldWrap".into(), should_wrap.into());
+    self.should_wrap = should_wrap;
```

Comment:
Thoughts on constructing meta during serialization of the asset graph for fields also directly stored on the dep, or do you think that will be too expensive?

Reply:
I think either is probably fine. I opted to do it like this as I think it's simpler to debug and understand. Rather than having a bunch of magic properties appear as it's serialised.